File: leetcode/lc-us/py/2530.py
import heapq
from heapq import heapify

from typing import List


class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        q = [-x for x in nums]
        heapify(q)
        res = 0
        while k:
            t = -heapq.heappop(q)
            res += t
            heapq.heappush(q, -((t + 2) // 3))
            k -= 1
        return res

File: leetcode/lc-us/cpp/2530.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    long long maxKelements(vector<int> &nums, int k) {
        priority_queue<int> heap;
        for (int x: nums) heap.push(x);
        long long res = 0;
        while (k--) {
            int t = heap.top();
            heap.pop();
            res += t;
            heap.push((t + 2) / 3);
        }
        return res;
    }
};